src/decoder.c

```diff
@@ -88,7 +88,6 @@ void cam_nplusone_decoder(struct decoder *d) {
   int sync, trigger;
   decoder_state oldstate = d->state;
 
-  disable_interrupts();
```

Comment:
No longer need critical regions since the whole thing is a critical region.

But since decode is now synchronous, change decode function to just take flags?
Also, we need to switch this to not being a single value, but a circular (dma provided) buffer with timestamps.  Integrate into the already-existing averaging array.